carbon-design-system/carbon-addons-iot-react
# fix(table): table toolbar action overflow truncation

### Merging into `master` to make it into 8.7

Closes #3198 

**Summary**

- forces text to truncate instead of wrap when using icons in the overflow menu
- hides the toolbar action overflow menu when all items are hidden
- hides toolbar actions that aren't in the overflow menu when they are hidden

**Change List (commits, features, bugs, etc)**

- change html structure/classes to cause long text to truncate when using icons
- update story to demo long text with and without icons
- update story to add demo all items as 'hidden' knob
- update tests for new structure

**Acceptance Test (how to verify the PR)**

- go to [with filtering, toolbarActions, and custom toolbar content](https://deploy-preview-3203--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-table-table--with-filters)
- open toolbar overflow menu
- confirm long text is truncated
- turn on "demo all toolbar overflow items hidden" knob
- confirm overflow menu and actions are hidden

**Regression Test (how to make sure this PR doesn't break old functionality)**

- confirm overflow menu works as expected
- check inline overflow menu actions ([this story, for example](https://deploy-preview-3203--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-table-table--table-example-with-create-save-views)) to confirm they still function and look as expected



**Things to look for during review**
